NAAM.md

```diff
@@ -0,0 +1,275 @@
+
+### IPNS Record Publication
+
+Naam publishes IPNS records via Publishers, which create IPNI-compliant advertisements that
+embed IPNS records and are signed by their identity. The produced advertisements consist of:
+
+* **`PreviousID`** - the optional link to the previous Naam advertisement. `nil` indicates no
+  previous advertisements.
+* **`Provider`** - the publisher peer ID.
+* **`Addresses`** - the array of multiaddrs over which the publisher can be reached.
+* **`Entries`** - the link to an `EntryChunk` IPLD node, as specified by IPNI, that contains a
+  single entry: the multihash of IPNS record key with `nil` as link to `Next`.
+* **`ContextID`** - fixed to UTF-8 bytes encoding of the value `/ipni/naam`.
+* **`Metadata`** - the bytes representation of Naam metadata, consisting of:
+    * the varint representation of `ipns-record` multicodec code, i.e. `0x0300`, as bytes,
+    * followed by the marshalled IPNS record.
+* **`IsRm`** - the boolean value specifying weather an IPNS entry is being added/updated,
+  i.e. `false`, or removed, i.e. `true`.
+* **`Signature`** - the advertisement signature calculated as specified by IPNI specification.
+
+#### `Entries`
+
+`Entires` in a typical advertisement are responsible for capturing the list of multihashes
+corresponding to the content hosted by a provider. Once ingested, the indexer nodes facilitate
+lookup of providers over find APIs via the same multihashes.
+
+Naam repurposes the same advertisement structure by specifically crafting entries of advertisement
+such that the indexer node find API can also be repurposed to lookup IPNS records. Naam forms a
+single `EntryChunk` IPLD node, that contains a single multihash calculated as:
+
+* The SHA-256 multihash of IPNS Record Key, i.e. `/ipns/<ipns-key>`.
+   * The multihash value is encoded according to the IPNI advertisement encoding. See [IPNI Specification](IPNI.md). 
+   * There are no restrictions on the concrete IPNS key format as long as it complies with the [IPNS routing record specification](https://github.com/ipfs/specs/blob/main/ipns/IPNS.md#routing-record).
+
+This approach allows the resolvers to deterministically calculate the multihash used for lookup when
+interacting with the indexer node find APIs.
+
+#### Context ID
+
+The field `ContextID` in a regular advertisement is used as a way to uniquely identify the metadata
+associated to a list of multihashes. Naam uses a fixed value, `/ipni/naam`, as context ID to assure
+that the metadata associated to the published advertisement for the same IPNI key is singular and
+uniquely identifiable.
+
+Using a fixed value as context ID also enables network indexers to quickly differentiate Naam
+advertisements from a typical content advertisement. This opens up future opportunities for
+optimisation when it comes to ingesting Naam advertisements. For example, indexers can pro-actively
+process IPNS records and offer bespoke APIs that understand IPNS record expiry/validity.
+
```

Comment:
The context ID should have some namespace prefix at the end (these could simply be monotonically increasing integers), e.g. `/ipni/naam/1` or whatever naming makes sense to the publisher. This will allow Naam differentiating Naam metadata from other content metadata, while also allowing different IPNS records on the same advertisement chain.